react/components/BuyButton/index.js

```diff
@@ -76,26 +77,49 @@ export class BuyButton extends Component {
       isOneClickBuy,
       onAddStart,
       onAddFinish,
+      orderFormContext,
     } = this.props
     this.setState({ isAddingToCart: true })
     onAddStart && onAddStart()
 
     let showToastMessage = null
+    
     try {
       const minicartItems = skuItems.map(this.skuItemToMinicartItem)
-
       const {
         data: { addToCart: linkStateItems },
       } = await addToCart(minicartItems)
 
-      const success =
-        linkStateItems &&
-        skuItems.filter(
-          skuItem => !!linkStateItems.find(({ id }) => id === skuItem.skuId)
+      let success = null
+      if (!linkStateItems) {
+        // minicart does not have link state implemented, calling graphql directly
+        // SHOULD BE ERASED ON THE NEAR FUTURE
```

Comment:
I don't think we need erase in the future. I think this is a good fallback to secure the independent of the components